src/app/rundown-view/components/minishelf/mini-shelf.component.ts

```diff
@@ -0,0 +1,97 @@
+  protected mediaDuration: number = 0
+  private configurationMediaPreviewUrl: string
+  private configurationServiceSubscription: Subscription
+  private executeActionSubscription: Subscription
+  private serverPostrollDuration: number = 0
+
+  constructor(
+    private readonly actionService: ActionService,
+    private readonly configurationService: ConfigurationService,
+    private readonly mediaStateService: MediaStateService
+  ) {}
+
+  public ngOnInit(): void {
+    this.configurationServiceSubscription = this.configurationService.getStudioConfiguration().subscribe((configuration: StudioConfiguration) => {
+      this.configurationMediaPreviewUrl = configuration.data.settings.mediaPreviewUrl
+      this.serverPostrollDuration = configuration.data.blueprintConfiguration.ServerPostrollDuration
+    })
+    void this.updateMediaDataDuration()
+  }
+
+  private async updateMediaDataDuration(): Promise<void> {
+    if (this.segment.metadata?.miniShelfVideoClipFile !== undefined) {
+      let media: Media | undefined = await this.mediaStateService.getMedia(this.segment.metadata?.miniShelfVideoClipFile)
+      if (!media || !media.duration) {
+        return
+      }
+      if (media.duration < this.serverPostrollDuration) {
+        this.mediaDuration = this.NaN
+      } else {
+        this.mediaDuration = media.duration - this.serverPostrollDuration
+      }
+    }
+  }
+
+  public ngOnChanges(changes: SimpleChanges): void {
+    if ('segment' in changes) {
+      void this.updateMediaDataDuration()
+    }
+  }
+
+  public ngOnDestroy(): void {
+    this.configurationServiceSubscription?.unsubscribe()
+    this.executeActionSubscription?.unsubscribe()
+  }
+
+  protected get mediaPreviewUrl(): string {
+    const url: string = `${this.configurationMediaPreviewUrl}/media/thumbnail/${this.segment.metadata?.miniShelfVideoClipFile}`
+    return this.configurationMediaPreviewUrl ? url : this.defaultAssetForThumbnail
+  }
+
+  public getSanitizedTitle(): string {
+    return this.segment.name
+      .replace(/<[^>]*>/g, '') // remove html tags
+      .replace(/(\r\n|\n|\r)/gm, ' ') // remove newlines
+      .replace(/_/g, ' ') // replace underscores with spaces;
+      .replace(/&nbsp;/g, ' ') // remove &nbsp;
+      .replace(/\s+/g, ' ') // remove extra spaces
+      .replace(/&amp;/g, '&') // remove &amp;
+      .replace(/&quot;/g, '"') // remove &quot;
+      .replace(/&apos;/g, "'") // remove &apos;
+      .trim() // and finally trim;
+  }
+
+  protected executeAction(): void {
+    if (!this.videoClipAction) {
+      return
+    }
+    this.executeActionSubscription = this.actionService.executeAction(this.videoClipAction.id, this.segment.rundownId).subscribe()
+  }
+
+  protected handleMissingImage(event: Event): void {
+    ;(event.target as HTMLImageElement).src = this.defaultAssetForThumbnail
+  }
+
+  protected readonly NaN = NaN
+  protected readonly isNaN = isNaN
```

Comment:
This seems to be instance variables. All instance variables should be a the top of the class so we have a single place for all instance variables and then we don't have to "hunt" for them in files we are not familliar with.

Also, the `NaN = NaN` seems redundant and I would prefer it got the deleted.
The same for `isNaN = isNaN`. You do use this one in the html, but if you move the check to the `timer` then you don't need this here.